src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs

```diff
@@ -815,5 +815,29 @@ internal static ImmutableArray<INamespaceSymbol> GetPublicSymbols(this Immutable
         {
             return symbol.GetSymbol<FunctionPointerTypeSymbol>();
         }
+
+        /// <summary>
+        /// Returns true if the method has a [AsyncMethodBuilder(typeof(B))] attribute. If so it returns the "B".
+        /// Validation of builder type B is left for elsewhere. This method returns B without validation of any kind.
+        /// </summary>
+        internal static bool HasAsyncMethodBuilder(this Symbol symbol, [NotNullWhen(true)] out object? builderArgument)
```

Comment:
>HasAsyncMethodBuilder [](http://example.com/codeflow?start=29&length=21)

Nit: feels like a better name for this is `HasAsyncMethodBuilderAttribute`, since it specifically won't return the builder type if the method has a Task-like type with no attribute.